### PR TITLE
Only convert the current message to a string once the last fragment has been received

### PR DESCRIPTION
**Pull request by aeaton-overleaf**
### The kind of change this PR does introduce

* [x] a bug fix
* [ ] a new feature
* [ ] an update to the documentation
* [ ] a code change that improves performance
* [ ] other

### Current behaviour

When a long message is sent as several frames (observed in Chrome v99, but not in Firefox v97), if the frame boundary occurs in the middle of a multibyte character then the combined message will be corrupted by the addition of an extra character at the point where the frames were concatenated with `+=` into a single string.

### New behaviour

By building up the current message as parts of a buffer, and only combining them once all the data has been received, the final UTF-8 string is no longer corrupted.

### CODE REVIEW

**Review by das7pad — COMMENTED** [review]
LGTM. Can you add an entry to the History.md file and bump the version to `0.9.19-overleaf-5`?

It would be good to get a second review on this from @briangough :)

**Review by briangough — APPROVED** [review]
:+1: good catch,  we've hit similar problems in other parts of the code when concatenating chunks from a stream.

**Pull request merged**
